http://sources.redhat.com/ml/binutils/2003-05/msg00739.html

2003-05-23  H.J. Lu <hongjiu.lu@intel.com>

	* ltmain.sh: Make symlink for shared library if needed.

opcodes/

2003-05-23  H.J. Lu <hongjiu.lu@intel.com>

	* Makefile.am (libopcodes_la_LIBADD): Use "-L../bfd -lbfd"
	instead of "../bfd/libbfd.la".
	* Makefile.in: Regenerated.

--- binutils/ltmain.sh.dso	2002-03-22 00:16:20.000000000 -0800
+++ binutils/ltmain.sh	2003-05-23 22:51:20.000000000 -0700
@@ -4413,6 +4413,10 @@ relink_command=\"$relink_command\""
       # LD_LIBRARY_PATH before the program is installed.
       $show "(cd $output_objdir && $rm $outputname && $LN_S ../$outputname $outputname)"
       $run eval '(cd $output_objdir && $rm $outputname && $LN_S ../$outputname $outputname)' || exit $?
+      if test -n "$linkname"; then
+        $show "(cd $output_objdir && $rm ../$linkname && $LN_S $output_objdir/$linkname ../$linkname)"
+        $run eval '(cd $output_objdir && $rm ../$linkname && $LN_S $output_objdir/$linkname ../$linkname)' || exit $?
+      fi
       ;;
     esac
     exit 0
--- binutils/opcodes/Makefile.am.dso	2003-05-19 08:23:00.000000000 -0700
+++ binutils/opcodes/Makefile.am	2003-05-23 22:46:29.000000000 -0700
@@ -283,7 +283,7 @@ disassemble.lo: disassemble.c $(INCDIR)/
 
 libopcodes_la_SOURCES =  dis-buf.c disassemble.c
 libopcodes_la_DEPENDENCIES = $(OFILES) ../bfd/libbfd.la
-libopcodes_la_LIBADD = $(OFILES) @WIN32LIBADD@ ../bfd/libbfd.la
+libopcodes_la_LIBADD = $(OFILES) @WIN32LIBADD@ -L../bfd -lbfd
 libopcodes_la_LDFLAGS = -release $(VERSION) @WIN32LDFLAGS@
 
 # libtool will build .libs/libopcodes.a.  We create libopcodes.a in
